stream/stream_file.c

```diff
@@ -332,6 +332,13 @@ static int open_f(stream_t *stream, const struct stream_open_args *args)
     setmode(p->fd, O_BINARY);
 #endif
 
+#if HAVE_SEEK_DATA
+    off_t first_data = lseek(p->fd, 0, SEEK_DATA);
```

Comment:
you need to conditionalise this for `write`, because newly created files are obviously empty and that wouldn't work.